integration/setup/flags.go

```diff
@@ -0,0 +1,172 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package setup
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+// flags are used to override flags set from cli with test setup option.
+type flags struct {
+	targetPort       *int
+	targetTLS        *bool
+	handler          *string
+	targetUnixSocket *bool
+	proxyAddr        *string
+	compatPort       *int
+	compatTLS        *bool
+	postgreSQLURL    *string
+	tigrisURL        *string
+}
+
+// ApplyOpts applies opts to the flags to override it.
+func (f *flags) ApplyOpts(tb testing.TB, opts map[string]any) *flags {
+	for k, v := range opts {
+		switch k {
+		case "target-port":
+			targetPort, ok := v.(int)
+			require.True(tb, ok, "%s is not int: %T", k, v)
+
+			f.targetPort = &targetPort
+		case "target-tls":
+			targetTLS, ok := v.(bool)
+			require.True(tb, ok, "%s is not bool: %T", v)
+
+			f.targetTLS = &targetTLS
+		case "handler":
```

Comment:
I remember there were some handler-related questions in this issue. How is it used now? Do we use `handler` for `target` only? How does it work in case of compat collections setup? Should we call it "target-handler"?.. Do we need this flag at all having in mind that there are flags for postgresql-url and tigris-url, so we already know if we work with pg or Tigris?..

Reply:
> How is it used now?

The handler still defaults to pg. This will change in next PR. validators are now only applied to target.

> Do we use handler for target only?

Yes for the validator, the handler is used only for target. The compat does not use validator since we know compat is always compatible with all provider.

> Should we call it "target-handler"?

I  like this suggestion, let me handle this in another PR. For example, we have test-integration-mongodb and which target is mongodb equivalent but handler is not applicable here because it's not `pg` or `tigris`. From my understanding handler is what we implement, so if we can't create listener for it, we shouldn't call it an handler. What do you think?

> Do we need this flag at all having in mind that there are flags for postgresql-url and tigris-url, so we already know if we work with pg or Tigris?

Great point, let me see if I can remove it fully. But let's handle the handler logic in next PR.